pkg/controllers/management/cloudcredential/controller.go

```diff
@@ -0,0 +1,62 @@
+package cloudcredential
+
+import (
+	"context"
+	"fmt"
+	"github.com/rancher/rancher/pkg/controllers/management/globalnamespacerbac"
+	corev1 "github.com/rancher/types/apis/core/v1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"strings"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+)
+
+type Controller struct {
+	cloudCredentials  corev1.SecretInterface
+	managementContext *config.ManagementContext
+}
+
+func Register(ctx context.Context, management *config.ManagementContext) {
+	cloudCredentials := management.Core.Secrets("")
+	m := Controller{
+		cloudCredentials:  cloudCredentials,
+		managementContext: management,
+	}
+	m.cloudCredentials.AddHandler(ctx, "management-cloudcredential-controller", m.ccSync)
+}
+
+func (n *Controller) ccSync(key string, cloudCredential *v1.Secret) (runtime.Object, error) {
+	if cloudCredential == nil || cloudCredential.DeletionTimestamp != nil {
+		return cloudCredential, nil
```

Comment:
could you point to the place where role binding is being removed?

Reply:
https://github.com/rancher/rancher/blob/7fe9051d857866bd70291a8f026634b76b9e5cd1/pkg/controllers/management/globalnamespacerbac/rbac_common.go#L142